altos-rust/port/cortex-m0/src/peripheral/usart/control.rs

```diff
@@ -250,43 +524,83 @@ impl CR2 {
 /// CR3
 // ------------------------------------
 
-#[derive(Copy, Clone)]
+/// Defines the possible HardwareFlowControl configurations for the Usart.
+/*  None: No hardware flow control
+ *  Rts: Request to Send enabled
+ *  Cts: Clear to Send enabled
+ *  All: Both are enabled.
+ */
+#[derive(Copy, Clone, Debug)]
 pub enum HardwareFlowControl {
     None,
-    /// Request to Send
     Rts,
-    /// Clear to Send
     Cts,
-    /// Both
-    RtsCts,
+    All,
 }
 
-#[derive(Copy, Clone)]
+/// Defines the base address variable for control register three.
+#[derive(Copy, Clone, Debug)]
 struct CR3 {
     base_addr: *const u32,
 }
 
+/// Implements the Register trait for Usartx_CR3.
+/// Stores base address for control register three, which is the address
+/// of the Usart being used to access this register. Uses the base address
+/// combined with the register offset to calculate the register address.
 impl Register for CR3 {
+    /* Sets the base address as the Usart address.
+     * Returns itself to the calling routine.
+     */
     fn new(base_addr: *const u32) -> Self {
         CR3 { base_addr: base_addr }
     }
 
+    /* Helper function to calcluate the address of CR3.
+     * Supplies the base address to the `addr()` Register routine.
+     * Used in conjunction with the 'mem_offset' function below.
+     */
     fn base_addr(&self) -> *const u32 {
         self.base_addr
     }
 
+    /* Helper function to calculate the address of CR3.
+     * Supplies the CR3 offset address back to the 'addr()' Register routine.
+     * Used in conjuction with the 'base_addr' function above.
+     */
     fn mem_offset(&self) -> u32 {
         CR3_OFFSET
     }
 }
 
+/// Function implementations for the Usartx_CR3.
+/// These functions are called from the wrapper functions defined
+/// for the Control struct.
 impl CR3 {
-    fn set_hardware_flow_control(&self, hfc: HardwareFlowControl) {
+    /* Uses bit 8 and 9 in CR3 to set the hardware flow control to None, Rts,
+     * Cts, All.
+     *      Bit 8 RTSE: RTS enable
+     *          0: RTS hardware flow control disabled
+     *          1: RTS output enabled, data is only requested when there is
```

Comment:
See, this information might be helpful on the enum or the doc comments on the wrapper function since the wrapper is our public interface